### DISCUSSION TITLE
Page switch and { wait: true }

### DISCUSSION THREAD

**Discussion by SKFrozenCloud** [lupas/nuxt-fire-vuexfire-example#17] [Q&A] [answered]
1. Why have you put `{ wait: true}` when binding? Doesn't it make it slower and also old data is not removed when unbinding.

`await bindFirestoreRef('countDocument', ref, { wait: true })`

2. When switching to a different page it has to also dispatch the bind function. It will unbind -> bind -> refetch the whole DB again...

Any solution to this? Is that why you have put `{ wait: true}` ?

**Comment by lupas** [comment, marked as answer]
Hey @SKFrozenCloud 
You answered your own question. We put wait: try exactly because of the issue in 2. So far haven't had an issue with it.

Let me know if you find a better solution to this issue :)

**Reply by SKFrozenCloud** [reply]
Hi @lupas ,

Ah that makes sense now. I also forgot I was using SSR and I didn't bind the data on server side... gave me some errors.
If I found a better solution instead of `{ wait: true }` I will let you know.

Could you also explain why you used `mounted` instead of `created` in `ssr-vuexfire` example?

```
async mounted() {
  try {
    await this.$store.dispatch('bindCountDocument')
  } catch (e) {
    console.error(e)
  }
},
```

**Reply by lupas** [reply]
As you can see on we get data on server side here:
https://github.com/lupas/nuxt-fire-vuexfire-example/blob/master/pages/ssr-vuexfire/index.vue#L12

And on client side here:
https://github.com/lupas/nuxt-fire-vuexfire-example/blob/master/pages/ssr-vuexfire/index.vue#L23

See the comments in the code ;-)

Since mounted() only gets called on the client-side and we only fetch on the server-side, this should make sense.

**Reply by lupas** [reply]
We probably could also use fetch for both but only unbind on server-side... haven't tested. Bear in mind that this example is quite old, I made it before fetch() was even an option... if you find a better way feel free to let me know =)

**Reply by SKFrozenCloud** [reply]
Thanks!

I will tinker around and see if that works.

**Reply by SKFrozenCloud** [reply]
Hi again,

So from what I have gathered, the `fetch(context)` or `fetch({ app, store })` is deprecated but is still kept for backwards compatibility.
The new fetch() does not take in any parameters and works **very differently** than the old fetch(context) does. Check out the links below.

Solution:

The asyncData(context) function works almost exactly like the old fetch(context) that is used in the examples.
I tested it and just changed it from fetch to asyncData and can confirm it works just as normal (not surprising when they should be almost the same).

Also I tested using `created` instead of `mounted` and found that it worked (better in my case) since `created` is fired after the asyncData and "after" the Vue instance have been created but before the `mounted`. I have not found any limitations or problems with using `created` instead of `mounted` but I do think there might be some limitations/problems.

https://nuxtjs.org/docs/2.x/concepts/nuxt-lifecycle
https://nuxtjs.org/blog/understanding-how-fetch-works-in-nuxt-2-12
